<Action id="68790" issue="44767" author="denyeart" type="comment" body=" ~jyellick  What&apos;s your opinion on how to best handle this error scenario so that an appropriate error message can be returned?" created="2020-04-09 16:42:43.0" updateauthor="denyeart" updated="2020-04-09 16:42:43.0"/>
<Action id="68792" issue="44767" author="jyellick" type="comment" body="I would be fine with removing the ability to have empty consortiums (thus allowing arbitrary channel creation), this is really only useful in a dev/test scenario.  Keep in mind, we&apos;ll need to make this change behind an orderer capability." created="2020-04-09 17:59:41.0" updateauthor="jyellick" updated="2020-04-09 17:59:41.0"/>
<Action id="68794" issue="44767" author="denyeart" type="comment" created="2020-04-09 20:00:34.0" updateauthor="denyeart" updated="2020-04-09 20:00:34.0"> <body><! CDATA  ~jyellick Actually I was suggesting a smaller change... just fail the channel creation earlier with a good message, if there are no members in the consortium, at this location:   https://github.com/hyperledger/fabric/blob/release-1.4/orderer/common/msgprocessor/systemchannel.go#L295   I think this smaller change would not need a new orderer capability... the behavior would be the same as before, it just fails earlier and with a better error message.  ></body> </Action>
<Action id="68796" issue="44767" author="jyellick" type="comment" body="We&apos;re both on the same page about what the change would be, and where it would be.  But, this still requires a capability change, since it affects a transaction validity decision at the orderer.  If you have two mixed leveled versions of orderers, one could apply the channel creation while the other rejects it.  In this particular scenario, because the transaction was generated via configtxgen, and it (inappropriately) includes the org in the readset, we are getting a failure.  It does not mean that this path always results in failure." created="2020-04-09 20:09:48.0" updateauthor="jyellick" updated="2020-04-09 20:09:48.0"/>
<Action id="68806" issue="44767" author="denyeart" type="comment" body="Ok, I don&apos;t think it is worth making a capability change just for this... I&apos;ll plan on just adding a Warning message at this location in the code, that would have been enough to help troubleshoot." created="2020-04-10 12:49:52.0" updateauthor="denyeart" updated="2020-04-10 12:49:52.0"/>
<Action id="70686" issue="44767" author="manish-sethi" type="comment" body=" ~denyeart  - assigning to you, as clearing the backlog." created="2020-11-16 18:57:54.0" updateauthor="manish-sethi" updated="2020-11-16 18:57:54.0"/>
<Action id="70769" issue="44767" author="denyeart" type="comment" body="https://github.com/hyperledger/fabric/pull/2149" created="2020-11-20 02:45:34.0" updateauthor="denyeart" updated="2020-11-20 02:45:34.0"/>
